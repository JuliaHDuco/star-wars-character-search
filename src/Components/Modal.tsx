import * as React from 'react';
import './styling/Modal.scss';

interface IModalProps {
  children: any;
  onHide: (e: React.MouseEvent) => void;
}

const Modal = ({children, onHide}: IModalProps) => {
  return (
    <div className='modal-wrapper'>
      <div className='modal'>
        <button onClick={onHide} />
        {children}
      </div>
    </div>
  );
}

export default Modal;
