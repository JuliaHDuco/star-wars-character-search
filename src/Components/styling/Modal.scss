@import '../../styles/mixins';

.modal-wrapper {
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  background-color: rgba(0, 0, 0, .5);
  display: flex;
  justify-content: center;
  align-items: center;

  .modal {
    width: 80%;
    background-color: #fff;
    color: #000;
    overflow: scroll;
    padding: 1em;

    button {
      @include cross(#000, 15px, 3px, relative);
      border: none;
    }
  }
}
