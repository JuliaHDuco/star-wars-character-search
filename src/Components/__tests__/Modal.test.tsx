import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Modal from '../Modal';

const onHideMock = jest.fn();
const children = <div>Hi</div>;

test('should match snapshot', () => {
  const component = renderer.create(
    <Modal onHide={onHideMock} children={children}/>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should call onHide', () => {
  const component = renderer.create(
    <Modal onHide={onHideMock} children={children} />
  );
  const button = component.root.findAllByType('button')[0];

  button!.props.onClick();
  expect(onHideMock.mock.calls.length).toBe(1);
});
