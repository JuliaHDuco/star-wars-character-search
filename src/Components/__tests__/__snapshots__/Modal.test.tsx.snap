// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="modal-wrapper"
>
  <div
    className="modal"
  >
    <button
      onClick={[MockFunction]}
    />
    <div>
      Hi
    </div>
  </div>
</div>
`;
